File: 791/custom-sort-string.py
# 將 s 裡的字母，照 order 的順序排序
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        s = list(s)
        # 因字串無法 .sort()，所以 s = list(s) 先轉成 list
        s.sort(key=lambda c: order.index(c) if c in order else -1)
        # 本來我是寫 s.sort(key=lambda c: order.index(c)) 但有些c不在order裡
        # 所以我寫得長一點：如果 c in order 就把 index 找出來用，不然就-1隨便放
        return ''.join(s)

File: 791/custom-sort-string.c
// 要照著字母順序 order 將 s 的字重新排序
// 不過有些字母不在order裡，那就隨便放
char table[256]={}; //字母序對照表
int cmp(const void *p1, const void *p2) { //照字母序比大小
    return table[*(char*)p1]-table[*(char*)p2];
}
char* customSortString(char* order, char* s) {
    for(int i=0; order[i]!=0; i++){
        table[order[i]]=i; //製作「字母序」的對照表
    }
    int N = strlen(s); //字串長度
    qsort(s, N, sizeof(char), cmp);
    return s;
}
